playground/cdk/CdkSample.AppHost/storage.module.bicep

```diff
@@ -1,77 +1,69 @@
-param principalId string
-
-@description('')
-param principalType string
-
-@description('')
-param storagesku string
-
-@description('')
-param locationOverride string
-
-
-resource storageAccount_1XR3Um8QY 'Microsoft.Storage/storageAccounts@2022-09-01' = {
-  name: toLower(take('storage${uniqueString(resourceGroup().id)}', 24))
-  location: locationOverride
-  tags: {
-    'aspire-resource-name': 'storage'
-  }
-  sku: {
-    name: storagesku
-  }
-  kind: 'StorageV2'
-  properties: {
-    accessTier: 'Hot'
-    networkAcls: {
-      defaultAction: 'Allow'
+resource storage 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+    name: take('storage${uniqueString(resourceGroup().id)}', 24)
+    kind: 'StorageV2'
+    location: locationOverride
+    sku: {
+        name: storagesku
+    }
+    properties: {
+        allowSharedKeyAccess: false
+        minimumTlsVersion: 'TLS1_2'
+        networkAcls: {
+            defaultAction: 'Allow'
+        }
+    }
+    tags: {
+        'aspire-resource-name': 'storage'
     }
-    minimumTlsVersion: 'TLS1_2'
-  }
 }
 
-resource blobService_vTLU20GRg 'Microsoft.Storage/storageAccounts/blobServices@2022-09-01' = {
-  parent: storageAccount_1XR3Um8QY
-  name: 'default'
-  properties: {
-  }
+resource blobs 'Microsoft.Storage/storageAccounts/blobServices@2023-01-01' = {
+    name: 'default'
+    parent: storage
 }
 
-resource roleAssignment_Gz09cEnxb 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  scope: storageAccount_1XR3Um8QY
-  name: guid(storageAccount_1XR3Um8QY.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe'))
-  properties: {
-    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')
-    principalId: principalId
-    principalType: principalType
-  }
+resource StorageBlobDataContributor_storage 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'StorageBlobDataContributor_storage')
+    properties: {
+        principalId: principalId
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')
+        principalType: principalType
+    }
+    scope: storage
 }
 
-resource roleAssignment_HRj6MDafS 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  scope: storageAccount_1XR3Um8QY
-  name: guid(storageAccount_1XR3Um8QY.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3'))
-  properties: {
-    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3')
-    principalId: principalId
-    principalType: principalType
-  }
+resource StorageTableDataContributor_storage 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'StorageTableDataContributor_storage')
```

Comment:
This doesn't look unique enough. Can we do what we had before?

```
  name: guid(keyVault_aMZbuK3Sy.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483'))
```

Reply:
The default value comes from https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/provisioning/Azure.Provisioning/src/Authorization/RoleAssignment.cs#L15 where we don't have enough information to create that value.  You'd need to explicitly set `Name` in your code.  `BicepFunction.CreateGuid` takes `params` so you should be be able to do it without dropping down to expressions.

Reply:
This is now fixed with the latest build.